### PR TITLE
Add install-salt-minion.sh

### PR DESCRIPTION
**Pull request by diegoakechi**
Install the salt-minion from the SUSE official Repository using
registration. 
This script can be used for provisioning machines or images that do not have a pre-installed salt-minion.

### CODE REVIEW

**Comment by juadk** [comment]
LGTM, I already use it, thanks.

File: salt/install-salt-minion.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        exit 0
+        ;; 
+    d) 
+        DEREGISTER="true"
+        ;;
+    r)
+        install_salt_minion "$OPTARG"
+        exit 0 
+        ;;
+    \?) echo "Invalid option -$OPTARG" >&2
```
Comment (ricardobranco777) [inline salt/install-salt-minion.sh]:
```suggestion
   *) echo "Invalid option -$OPTARG" >&2
```

File: salt/install-salt-minion.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    # Check SLE version
+    source /etc/os-release
+
+    SUSEConnect -r "$regcode"
+
+    # Register the modules accordingly with the SLE version.
+    if [[ $VERSION == *"12"* ]]; then    
+      SUSEConnect -p sle-module-adv-systems-management/12/x86_64
+    elif [[ $VERSION == *"15"* ]]; then
```
[the remark this answers is not in the document]
Comment (diegoakechi) [reply salt/install-salt-minion.sh]: Makes sense @ricardobranco777. Thanks!

File: salt/install-salt-minion.sh (hunk 1 above)
Comment (diegoakechi) [reply salt/install-salt-minion.sh]: @ricardobranco777 I am using `?` as recommended on the getopts man page. Any reason where `*` would be better here?

Comment (ricardobranco777) [reply salt/install-salt-minion.sh]: The `getopts` example uses `?)` but `*)` helps to catch unhandled options specified in the option string.  It's very much like using `default` in the C version of `getopt`.

**Pull request merged**